### DISCUSSION TITLE
Support for Retry with LINEAR_BACKOFF

### DISCUSSION THREAD

**Discussion by venkag** [Netflix/conductor#2710] [Ideas]
Retry has support for two different backoff methods:

```
n - Configuration Parameter that defines "retryDelaySeconds"
i - Retry count (a given iteration number actually). If configuration parameter "retryCount" is 3, i runs from 0..2.
```

1. FIXED - Every retry is performed with a delay of `n` number of seconds (i is not accounted for).
2. EXPONENTIAL_BACKOFF - Every retry is performed with a `n* (2^i)` number of seconds.

I would like to add a BACKOFF method that sits between `FIXED` and `EXPONENTIAL` and name it as `LINEAR_BACKOFF`. With this backoff method, every retry will be performed with a delay of `n * (s*i)` number of seconds.

A new configuration parameter is added to be able to control the slope of the line.
```
s - slope or scale factor that controls how slow or fast the delay grows linearly.
```

Given, this is requires simple change and does not effect any other feature (or default configurations etc.), a PR (#2699) was already raised.

cc: @aravindanr @apanicker-nflx @kishorebanala

**Comment by venkag** [comment]
## Graphical representation of the retry behavior

<img width="1175" alt="Screen Shot 2022-01-24 at 9 56 24 PM" src="https://user-images.githubusercontent.com/14304605/150902601-4fa5d1be-2df8-4b2d-bc45-2ee7bab51fe5.png">

**Comment by venkag** [comment]
Updating the thread on the status of the PR: #2699 is merged. Thanks @apanicker-nflx

**Reply by apanicker-nflx** [reply]
Released in v3.5.0
